src/components/widget/RawWidget.js

```diff
@@ -322,7 +322,7 @@ class RawWidget extends Component {
                             type="text"
                             ref={c => this.rawWidget = c}
                             className="input-field js-input-field"
-                            value={widgetData[0].value}
+                            defaultValue={widgetData[0].value}
```

Comment:
@damianprzygodzki , @Dunkat 
random thought (pls ignore if does not apply): wondering if making the caret position as part of the state would do the trick....